features/cellular/framework/AT/ATHandler.cpp

```diff
@@ -120,8 +126,15 @@ bool ATHandler::get_debug() const
 
 ATHandler::~ATHandler()
 {
+    ScopedLock <ATHandler> lock(*this);
     set_file_handle(NULL);
 
+    // cancel possible oob event from queue and wait if oob is ongoing
+    _queue.cancel(_event_id);
```

Comment:
If the `cancel` reports success, then you need to set `oob_pending` to false, no? Maybe make this conditional on `oob_queued`, to avoid stale `event_id`.

(Could `_oob_queued` be replaced by a null `event_id` check? Don't need separate variables?)

Reply:
As cancel is void function we decided not to call it. 
Replaced _oob_queued  and _oob_pending with _event_id.
Force pushed changes.

Reply:
equeue should be handling stale event_ids.

Each event gets its own generation count which is packed into the id. If the generation count doesn't match `equeue_cancel` is a noop.

`equeue_cancel(0)` is also a noop, much like `free(NULL)`.